cypress/integration/Homework/calculator.spec.js

```diff
@@ -0,0 +1,408 @@
+/// <reference types="cypress" />
+
+import Calculator from '../../page_objects/calculator.js'
+
+var calculator = new Calculator
+
+// Build version names for the forEach cycle.
+var buildVersions = ["Prototype", "1", "2", "3", "4", "5", "6", "7", "8", "9"];
+//
+
+// Numbers for the TS1, TS2, TS4 and TS6 phases. Can be a ' ', but no symbols.
+var firstNumber = 8.5;
+var secondNumber = 2.2;
+//
+
+// Symbols for the TS5 phase. One of them must be a symbol, not a ' '.
+var firstSymbol = ' ';
+var secondSymbol = 'P';
+//
+
+// Variables for error messages.
+var errorMessage;
+var divideByZeroError = 'Divide by zero error!';
+//
+
+// Defining the error message based on the selected symbols.
+if (typeof firstSymbol === "string" && firstSymbol != ' ') {
+    errorMessage = 'Number 1 is not a number';
+} else if (typeof secondSymbol === "string") {
+    errorMessage = 'Number 2 is not a number';
+}
+//
+
+buildVersions.forEach((buildVersion) => {
+
+    [true, false].forEach((integersOnly) => {
+
+        // Operations for answer checking.
+        var addResultNumber = +firstNumber + +secondNumber;
+        var subtractResultNumber = firstNumber - secondNumber;
+        var multiplyResultNumber = firstNumber * secondNumber;
+        var divideResultNumber = firstNumber / secondNumber;
+        var concatenateResultNumber = firstNumber.toString() + secondNumber.toString();
+        //
+
+        // If the Integers only checkbox is checked then it makes alterations for the following answers.
+        if (integersOnly) {
+            addResultNumber = addResultNumber | 0;
+            subtractResultNumber = subtractResultNumber | 0;
+            multiplyResultNumber = multiplyResultNumber | 0;
+            divideResultNumber = divideResultNumber | 0;
+        }
+        //
+
+        describe('TS1 (Build - ' + buildVersion + '): Checking if the "First number" field exists and the user is able to input data', () => {
+        
+            before("Visits the initial page", () => {
+                calculator.visitInitialPage()
+                calculator.getBuildDropDownList().select(buildVersion)
+            })
+        
+            it("TS1.1: Checks if the 'First number' field exists on the 'Add' operation and writes " + firstNumber + " into the field", () => {
+                calculator.getOperationDropDownList().select('Add')
```

Comment:
It could be unit tests but not e2e. Simple test adding two numbers already covers this.

Reply:
Same with many tests below.

Reply:
I do agree with this. Got a little to carried away with writing these tests. I will remove them. :)

Reply:
Looking at it from the e2e perspective, the only viable tests are TS4, TS5 and TS6, am I correct? TS1 and TS2 sections only overlook one input field and TS3 evaluates only the 'Calculate' section.

Reply:
Didn't look through every test. Simple tests for sum, multiplication and similar should do the thing.